share/config.py

```diff
@@ -264,3 +272,107 @@ def parse_config(config_yaml: str) -> Config:
         conf.add_input(current_input)
 
     return conf
+
+
+def aws_sm_expander(config_yaml: str) -> str:
+    needed_secret_arns = {}
+    needed_secret_keys = {}
+
+    re_pattern = r"arn:aws:secretsmanager:(?:[^:]*):(?:[^:]*):secret:(?:[^\n]*)"
```

Comment:
`[^\n]` is this for not matching a new line?
I would rather go for an ungreedy pattern/no multiline flag if python has support for those

Reply:
totally agree, I will look for something that will match up to the double quotes and an optional last part (for the json key)

Reply:
how does this look? `"arn:aws:secretsmanager:(?:[^:]*):(?:[^:]*):secret:(?:[^"|']*)(?::[^"|']*)?"`

after the `secret` it will match until it hits double quotes or single quotes, or it can have an optional `:JSON_KEY"` group, that will still match until hits single/double quotes.

Reply:
> 

i realized it's still greedy, still not what we want. working on it...